firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FeedbackActivity.java

```diff
@@ -28,20 +31,42 @@ public class FeedbackActivity extends AppCompatActivity {
 
   public static final String RELEASE_NAME_EXTRA_KEY =
       "com.google.firebase.appdistribution.FeedbackActivity.RELEASE_NAME";
-  public static final String SCREENSHOT_EXTRA_KEY =
-      "com.google.firebase.appdistribution.FeedbackActivity.SCREENSHOT";
+  public static final String SCREENSHOT_FILENAME_EXTRA_KEY =
+      "com.google.firebase.appdistribution.FeedbackActivity.SCREENSHOT_FILE_NAME";
 
   private FeedbackSender feedbackSender;
   private String releaseName;
-  private Bitmap screenshot;
+  @Nullable private Bitmap screenshot;
 
   @Override
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     releaseName = getIntent().getStringExtra(RELEASE_NAME_EXTRA_KEY);
-    screenshot = getIntent().getParcelableExtra(SCREENSHOT_EXTRA_KEY);
+    screenshot = readScreenshot(getIntent().getStringExtra(SCREENSHOT_FILENAME_EXTRA_KEY));
```

Comment:
How about creating a thumbnail in `ScreenshotTaker` (in addition to writing the full resolution file)? It should be possible to pass the Thumbnail directly via the intent so it can be displayed in this activity. The full resolution file can then be read in `FirebaseAppDistributionTesterApiClient` without getting read and decompressed here - and then recompressed in the client...

Reply:
Any displayable bitmap is really too big to be passing via Intent, since they should be kept to "a few kb".

However, I've updated FeedbackAction to read a scaled thumbnail from the png file, which will reduce the memory footprint when starting the action.